src/resolver/resolver.zig

```diff
@@ -2542,25 +2542,20 @@ pub const Resolver = struct {
         var path = dir_info_uncached_path_buf[0.._path.len];
 
         bufs(.dir_entry_paths_to_resolve)[0] = (DirEntryResolveQueueItem{ .result = top_result, .unsafe_path = path, .safe_path = "" });
-        var top = Dirname.dirname(path);
+        var top = std.fs.path.dirname(path) orelse &([_]u8{});
```

Comment:
☝️ Instead of `Dirname.dirname` using `std.fs.path.dirname` with a consistent result (avoids the `null`).